### DISCUSSION TITLE
How can I debug tailwind server not attaching?

### DISCUSSION THREAD

**Discussion by danielo515** [LunarVim/LunarVim#3638] [Q&A]
Hello.
I have a problem with tailwind lsp. It sometimes attaches to the right buffers, and some times it just decides to not start, and I can't understand why.
Checking the LSP log does not reveal any problem related to tailwind, and there are just some warnings. 
How can I debug this? Is there a way I can force tailwind-lsp to attach to all the buffers I care about?

**Comment by danielo515** [comment]
So I just surrender and manually added the tailwind configuration for all the filetypes I am interested on:

```lua
  lspManager.setup(
    "tailwindcss",
    { filetypes = { "rescript", "reason", "typescriptreact", "javascript", "javascriptreact" } }
  )
```

Probably in the future I will have to add more manually, but at least this way I'm sure what config is being applied, because I can't find any other reliable way of seeing what logic is being applied here

**Comment by danielo515** [comment]
I still will love to have some instructions about how to debug it. The tailwind-css server is slow as hell, and I would love to know why

**Comment by LostInTheLogs** [comment]
You can find the default filetypes [here](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#tailwindcss)

for debugging i'd first check LspInfo, then the lsp log and then autogenerated ftplugin files - you can search for them using `leader L f` (e.g. typescript.lua)

**Comment by danielo515** [comment]
I just found that, I ran `LspSettings tailwindcss`, which created an empty JSON file, which I never filled. This may be the reason why some of my lsp settings are being ignored, the JSON file seems to take precedence.
